[TensorPipe Agent] Improve Response Error Message on Agent Shutdown

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38818 [TensorPipe Agent] Improve Response Error Message on Agent Shutdown**

Standardizes the error message when a response is attempted after the
agent has shut down.

Differential Revision: [D21670706](https://our.internmc.facebook.com/intern/diff/D21670706/)

osalpekar committed:
[TensorPipe Agent] Improve Response Error Message on Agent Shutdown

Standardizes the error message when a response is attempted after the
agent has shut down.

Differential Revision: [D21670706](https://our.internmc.facebook.com/intern/diff/D21670706/)

[ghstack-poisoned]

osalpekar commented on torch/csrc/distributed/rpc/tensorpipe_agent.cpp:
```diff
@@ -219,7 +219,13 @@ void TensorPipeAgent::sendCompletedResponseMessage(
+        "tried to respond to message with id ",
+        messageId,
+        " but RPC is no longer running on this node.");
+    LOG(WARNING) << err;
```
The equivalent check in `TensorPipeAgent::send` throws a runtime error here. Should we do the same here?

jiayisuse replied:
send() is user thread while this is rpc server thread, may be better no to throw exception :)

mrshenli commented on torch/csrc/distributed/rpc/tensorpipe_agent.cpp:
```diff
@@ -219,7 +219,13 @@ void TensorPipeAgent::sendCompletedResponseMessage(
+    auto err = c10::str(
+        "Node ",
+        RpcAgent::getWorkerInfo().id_,
+        "tried to respond to message with id ",
```
missing a space before "tried"?

osalpekar committed:
Update on "[TensorPipe Agent] Improve Response Error Message on Agent Shutdown"


Standardizes the error message when a response is attempted after the
agent has shut down.

Differential Revision: [D21670706](https://our.internmc.facebook.com/intern/diff/D21670706/)

[ghstack-poisoned]
